src/main/java/slimeknights/tconstruct/gadgets/entity/ThrowableItem.java

```diff
@@ -0,0 +1,28 @@
+package slimeknights.tconstruct.gadgets.entity;
+
+import net.minecraft.client.util.ITooltipFlag;
+import net.minecraft.item.ItemStack;
+import net.minecraft.item.SnowballItem;
+import net.minecraft.util.text.ITextComponent;
+import net.minecraft.world.World;
+import net.minecraftforge.api.distmarker.Dist;
+import net.minecraftforge.api.distmarker.OnlyIn;
+import slimeknights.mantle.util.TranslationHelper;
+import slimeknights.tconstruct.gadgets.TinkerGadgets;
+
+import javax.annotation.Nullable;
+import java.util.List;
+
+public abstract class ThrowableItem extends SnowballItem {
```

Comment:
I feel throwballs should also extend this class, or its a bit pointless abstraction. Especially since you did not make use of the optional tooltip.